### PR TITLE
Deadchat control alerts ghosts when they can input another command

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

Makes it so deadchat control alerts ghosts when they can input another command. Also rounds the seconds values in the to_chat messages for deadchat control up, so they don't have a decimal point.

dreamseeker_nPhiyvxl7a

## Why It's Good For The Game

You don't have to keep trying to input something to see if the cooldown has expired yet. Removing the decimal point looks cleaner.

## Changelog

:cl:
qol: Deadchat control on anarchy mode will now alert you when you can enter another command
/:cl:

### CODE REVIEW

File: code/datums/components/deadchat_control.dm [hunk 1]
```diff
@@ -191,9 +194,9 @@
 	examine_list += span_notice("[A.p_theyre(TRUE)] currently under deadchat control using the [(deadchat_mode & DEMOCRACY_MODE) ? "democracy" : "anarchy"] ruleset!")
 
 	if(deadchat_mode & DEMOCRACY_MODE)
-		examine_list += span_notice("Type a command into chat to vote on an action. This happens once every [input_cooldown * 0.1] seconds.")
+		examine_list += span_notice("Type a command into chat to vote on an action. This happens once every [CEILING(input_cooldown * 0.1, 1)] second\s.")
 	else if(deadchat_mode & ANARCHY_MODE)
-		examine_list += span_notice("Type a command into chat to perform. You may do this once every [input_cooldown * 0.1] seconds.")
+		examine_list += span_notice("Type a command into chat to perform. You may do this once every [CEILING(input_cooldown * 0.1, 1)] second\s.")
```
Comment (LemonInTheDark) [inline code/datums/components/deadchat_control.dm:199]: Why all these CEILINGS? a cooldown of 0.1 seconds would be fine here

Comment (cacogen) [reply code/datums/components/deadchat_control.dm]: ok, i got rid of them on everything except the message when you try to submit a command too early

File: code/datums/components/deadchat_control.dm [hunk 2]
```diff
@@ -63,13 +63,16 @@
 		return
 
 	if(deadchat_mode & ANARCHY_MODE)
+		if(!source || !source.ckey)
+			return
 		var/cooldown = ckey_to_cooldown[source.ckey] - world.time
 		if(cooldown > 0)
-			to_chat(source, span_warning("Your deadchat control inputs are still on cooldown for another [cooldown * 0.1] seconds."))
+			to_chat(source, span_warning("Your deadchat control inputs are still on cooldown for another [CEILING(cooldown * 0.1, 1)] second\s."))
```
Comment (LemonInTheDark) [inline code/datums/components/deadchat_control.dm:70]: Honestly the same question still applies here, why round this up?

Comment (cacogen) [reply code/datums/components/deadchat_control.dm:70]: Having a decimal point in  a seconds value looks like a mistake. But if you round down it'll say "0 seconds." when the value is like 0.2 seconds. Which also looks like a mistake. If it were really 0 seconds, they'd already be unlocked.

Comment (LemonInTheDark) [reply code/datums/components/deadchat_control.dm:70]: you don't think a delay of 1.5 seconds is ok?

Comment (cacogen) [reply code/datums/components/deadchat_control.dm:70]: Pretty sure it's just the one decimal place but it does look wrong to show that to the player, yeah.

Comment (LemonInTheDark) [reply code/datums/components/deadchat_control.dm:70]: I'm not sure I totally agree, but alright

**Pull request merged**